packages/dds/tree/src/test/feature-libraries/chunked-forest/codec/schemaBasedEncode.spec.ts

```diff
@@ -229,6 +233,7 @@ describe("schemaBasedEncoding", () => {
 				() => fail(),
```

Comment:
You modified getNodeEncoder, adding some new logic to it. You should update its test suite to use and test that logic.

Currently the only change toy made to the test here is to pass in undefined for the incrementalEncoder: that is leaving the case where it's not undefined as well as the actual selection and use of incrementalFieldEncoder without test coverage.